### ISSUE TITLE
heatmap deos not converge

### ISSUE THREAD

**Issue by DeeperDeeper** [xingyizhou/CenterNet#252]
Hey @xingyizhou 
I am implementing CenterNet using DLA model from [https://github.com/Stick-To/CenterNet-tensorflow/blob/master/CenterNet.py](DLA). 
I am using those functions to generate the heatmap and looks good, however, it did not converge. I have used the same loss weights and the same optimizer. I have tried also focal loss and LeastMeanSquare but both of them does not converge!! I see also if two objects centers are very close, we can not get two sharp peaks!!
```
def get_single_heatmap(heatmap, index, joint, sigma, height, width, stride):
    """
    Create a single heatmap corresponding to the joint (index)
    Parameters
    ----------
    heatmap : numpy array
        holder of the heat map
    index : int
        index of the heatmap
    joint : list
        joint coordinate [x, y]
    sigma : float
        sigma value for gaussian heatmap
    height : int
        image height
    width : int
        image width
    stride : int
        Scale factor between the input size and the target size, ex. 2, 4, 8

    Returns
    -------

    """

    center_x, center_y = int(joint[0] // stride), int(joint[1] // stride)

    for y in range(height):
        for x in range(width):
            d2 = (x - center_x) ** 2 + (y - center_y) ** 2
            exponent = d2 / (2.0 * sigma ** 2)
            v = math.exp(-exponent)
            if v < 1e-4:
                continue
             # If two objects overlaps, we take the max value
            prev_value = heatmap[y, x, index]
            heatmap[y, x, index] = np.maximum(v, prev_value)
            if heatmap[y, x, index] > 1.0:
                heatmap[y, x, index] = 1.0
```

```
def create_heatmap_center_point(centers, bboxes, catgs, width, height, stride, sigma, num_classes):
    """
    Create a gaussian heatmap for bounding box center. Here we have num_classes channels
    in the heatmap (channels = 80 in coco dataset).
    Parameters
    ----------
    centers : numpy array
        bbox center
    bboxes: numpy array
        list of bouding boxes in one image
    catgs : numpy array
        category of objects in one image
    width : int
        heatmap width
    height : int
        heatmap height
    stride : int
        stride between input and output size (here is 4,  512 -> 128)
    sigma : float
        sigma radius default value, here we use dynamic sigma based on the area of the 
        bbox
    num_classes : int
        number of classes or categories, in pascal voc= 20 classes
    Returns
    -------

    """
    heatmap = np.zeros((width, height, num_classes))
    for center, bbox, idx in zip(centers, bboxes, catgs):
        sigma_radius = gaussian_radius(bbox)
        sigma_value = np.maximum(sigma, sigma_radius)
        get_single_heatmap(heatmap, idx, center, sigma_value, height, width, stride)

    return heatmap
```

![Screenshot from 2019-08-05 16-12-10](https://user-images.githubusercontent.com/13174842/62492244-02b97300-b79c-11e9-91e7-d88bfbdcc760.png)

![Screenshot from 2019-08-05 16-12-24](https://user-images.githubusercontent.com/13174842/62492249-077e2700-b79c-11e9-91f9-9764b92c9bb7.png)

**Comment by xingyizhou** [comment]
I am not able to debug new implementations. Good luck.

**Comment by DeeperDeeper** [comment]
@xingyizhou, I don't mean that you debug my implementation, just I need some hints because you maybe had such problem. I try to overfit one image using only the `focal loss` but it does not converge. However, with the same heatmap, it converges easily `mean_square_error` loss with lr=0.002 and using
